web/html/src/manager/content-management/project/project.js

```diff
@@ -1,22 +1,122 @@
 // @flow
-import React, { Component } from 'react';
+
+// $FlowFixMe  // upgrade flow
+import React, { useState, useEffect } from 'react';
 import { hot } from 'react-hot-loader/root';
+import {TopPanel} from "../../../components/panels/TopPanel";
+import Sources from "../shared/components/panels/sources/sources";
+import PropertiesEdit from "../shared/components/panels/properties/properties-edit";
+import Build from "../shared/components/panels/build/build";
+import EnvironmentPath from "../shared/components/panels/environment-path/environment-path";
+
+import {handleSourcesChange, handleBuild} from "../shared/state/project/project.state";
+import { showErrorToastr, showSuccessToastr } from 'components/toastr/toastr';
+import Filters from "../shared/components/panels/filters/filters";
+import _isEmpty from "lodash/isEmpty";
+
+import "./project.css";
+import {DeleteDialog} from "components/dialog/DeleteDialog";
+import {ModalButton} from "components/dialog/ModalButton";
+import useProjectActionsApi from '../shared/api/use-project-actions-api';
 
 type Props = {
+  project: Object,
+  wasFreshlyCreatedMessage?: string,
 };
 
-type State = {
-};
+//TODO:  React 16 -> Default error handling implement
+
+const Project = (props: Props) => {
+
+  const [project, setProject] = useState(props.project)
+  const {onAction} = useProjectActionsApi({ projectId: project.properties.label });
+
 
-// eslint-disable-next-line react/prefer-stateless-function
-class Project extends Component<Props, State> {
-  render() {
-    return (
-      <div>
-        Project Detail
+  useEffect(()=> {
+    if(props.wasFreshlyCreatedMessage) {
+      showSuccessToastr(props.wasFreshlyCreatedMessage)
+    }
+  }, [])
+
+  const editedStates = ["0","2","3"];
+  const statesDesc = {
+    "0": "added",
+    "1": "built",
+    "2": "edited",
```

Comment:
This will be removed on the sources UI PR, all the logic will be implemented on the server side. If you prefer I can remove all the sources logic from this PR. 

Anyway, I think we will still need the state Edited for filters in case we reuse the same enum.

We also talked about the fact to give some visual feedback that a project needs to be rebuilt when an "mgr-sync" is run and a source gets new packages, we could use that edited (yellow) state to represent that.